Extension/.scripts/common.ts

```diff
@@ -0,0 +1,290 @@
+    return Promise.all(stdio.filter("Would remove").map((s) => filepath.exists(s.replace(/^Would remove /, ''), $root)).filter(p => p));
+}
+
+export async function rimraf(...paths: string[]) {
+    const all = [];
+    for (const each of paths) {
+        if (!each) {
+            continue;
+        }
+        if (await filepath.isFolder(each)) {
+            verbose(`Removing folder ${red(each)}`);
+            all.push(rm(each, {recursive: true, force: true}));
+            continue;
+        }
+        verbose(`Removing file ${red(each)}`);
+        all.push(rm(each, {force: true}));
+    }
+    await Promise.all(all);
+}
+
+export async function mkdir(filePath: string) {
+    const [fullPath, info] = await filepath.stats(filePath, $root);
+    if (info) {
+        if (info.isDirectory()) {
+            return fullPath;
+        }
+        throw new Error(`Cannot create directory '${filePath}' because thre is a file there.`);
+    }
+
+    await md(fullPath, { recursive: true });
+    return fullPath;
+}
+
+export const glob: (pattern: string, options?: IOptions | undefined) => Promise<string[]> = promisify(globSync);
+
+export async function write(filePath: string, data: Buffer|string) {
+    await mkdir(dirname(filePath));
+
+    if (await filepath.isFile(filePath)) {
+        const content = await readFile(filePath);
+        if (is.string(data)) {
+            // if we're passed a text file, we should match the line endings of the existing file.
+            const textContent = content.toString();
+
+            // normalize the line endings to the same as the current file.
+            data = textContent.indexOf('\r\n') > -1 ? data.replace(/\r\n|\n/g, '\r\n') :  data.replace(/\r\n|\n/g, '\n');
+
+            // if the text content is a match, we don't have to change anything
+            if (textContent === data) {
+                verbose(`Text file at '${filePath}' is up to date.`);
+                return;
+            }
+        } else {
+            // if the binary content is a match, we don't have to change anything
+            if (content.equals(data)) {
+                verbose(`File at '${filePath}' is up to date.`);
+                return;
+            }
+        }
+    }
+
+    verbose(`Writing file '${filePath}'`);
+    await writeFile(filePath, data);
+}
+
+export async function updateFiles(files: string[], dest: string| Promise<string>) {
+    const target = is.promise(dest) ? await dest : dest;
+    await Promise.all(files.map(async (each) => {
+        const sourceFile = await filepath.isFile(each, $root);
+        if (sourceFile) {
+            const targetFile = resolve(target, each);
+            await write(targetFile, await readFile(sourceFile));
+        }
+    }));
+}
+
+export async function go() {
+    if (require.main) {
+        // loop thru the args and pick out the first non --arg and remove it from the $args and set $cmd
+        for (let i = 0;i < $args.length;i++) {
```

Comment:
Does the eslint formatter not run on this file? There are no spaces after `;`?

Reply:
There was no rule to insert spaces. I added a new rule.